### PR TITLE
Unionlabs split tendermint

### PR DESCRIPTION
**Pull request by benluelo**
- split tendermint light client types into their own crate
- split berachain light client types into their own crate (i would do this in a separate PR but the berachain light client depends on the tendermint types, if this wasn't also split out it would result in a circular dependency)

### CODE REVIEW

File: light-clients/berachain-light-client/Cargo.toml [hunk 1]
```diff
@@ -32,11 +32,13 @@ tendermint-light-client = { workspace = true, features = ["library"] }
 tendermint-verifier     = { workspace = true }
 thiserror               = { workspace = true }
 
-ethereum-verifier  = { workspace = true }
-hex-literal        = { workspace = true }
-ics008-wasm-client = { workspace = true }
-ics23              = { workspace = true }
-unionlabs          = { workspace = true }
+berachain-light-client-types.workspace  = true
```
Comment (aeryz) [inline light-clients/berachain-light-client/Cargo.toml:35]: let's keep `{ workspace = true }` convention

File: light-clients/berachain-light-client/Cargo.toml [hunk 2]
```diff
@@ -32,11 +32,13 @@ tendermint-light-client = { workspace = true, features = ["library"] }
...
+ethereum-verifier                       = { workspace = true }
+hex-literal                             = { workspace = true }
+ics008-wasm-client                      = { workspace = true }
+ics23                                   = { workspace = true }
+tendermint-light-client-types.workspace = true
```
Comment (aeryz) [inline light-clients/berachain-light-client/Cargo.toml:40]: let's keep `{ workspace = true }` convention

File: light-clients/berachain-light-client/types/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[package]
+edition = "2021"
+name    = "berachain-light-client-types"
+version = "0.1.0"
+
+[dependencies]
+protos    = { workspace = true, features = ["union+ibc+lightclients+berachain+v1"] }
+serde     = { workspace = true, features = ["derive"] }
+tendermint-light-client-types={workspace = true}
```
Comment (aeryz) [inline light-clients/berachain-light-client/types/Cargo.toml:9]: fmt plis

**Review by aeryz — DISMISSED** [review]

File: light-clients/berachain-light-client/types/Cargo.toml (hunk 3 above)
Comment (benluelo) [reply light-clients/berachain-light-client/types/Cargo.toml:9]: dude i did taplo broke idk why

File: light-clients/berachain-light-client/Cargo.toml (hunk 2 above)
Comment (benluelo) [reply light-clients/berachain-light-client/Cargo.toml:40]: im going to push a formatter change once all of this is done to fix this automatically for us, it's too much work to automatically maintain

**Review by aeryz — APPROVED** [review]

**Pull request merged**